hibernate-core/src/main/java/org/hibernate/query/spi/QueryOptions.java

```diff
@@ -161,6 +161,15 @@ default Limit getEffectiveLimit() {
 		return explicit != null ? explicit : Limit.NONE;
 	}
 
+	/**
+	 * Determine if the query execution has to be considered by the {@link org.hibernate.stat.Statistics}.
+	 *
+	 * @return true if the query execution has to be added to the {@link org.hibernate.stat.Statistics}, false otherwise.
+	 */
+	default boolean hasQueryExecutionToBeAddedToStatistics() {
```

Comment:
we are adding only query generated from hql

Reply:
Make sense, but then I'd rather see this on `ExecutionContext` as the mechanism is similar to `getQueryIdentifier` and return true by default. In case of a CRITERIA query source, you can return false

Reply:
it should return false also for loading object, so I'm adding the method to `ExecutionContext` as you suggest but with a default of `false`